### ISSUE TITLE
[FEATURE] customise refit

### ISSUE THREAD

**Issue by koaning** [koaning/scikit-lego#210]
At the moment `GridSearchCV` works by selecting the model that performs best on the `refit` metric on the test set. It doesn't offer a way for you to; 

- score different rows of data less heavily than others
- include not just the `mean_test_refit` score but also the `std_test_refit` score
- allow another holdout set to be used in determining the best outcome 

These can be valid concerns. For example; population drift can occur and you might want to make sure that you are not over-fitting on data that is too old. 

The goal of this thread is. to check what tools we might make that 1.) fit the sklearn pipeline and 2.) potentially solve some of these issues. The focus needs to remain on the user interface and it might very well be that we solve these problems with different solutions instead of one solution to solve them all.

**Comment by koaning** [comment]
After thinking about it (and while eating a portuguese cake) I came to a proposal for this. In general we'd be interested in metrics that we supply ourselves and maybe we're interested in selecting a set of models. We might go for a hindsight selector ... but maybe we need ... 

## ParetoGridSearch 

Here's the idea. There are many models that are tried. Suppose we want both precision and recall to be high. 

![image](https://user-images.githubusercontent.com/1019791/73679260-56a90400-46ba-11ea-922a-903bdb61c567.png)

Then we have an efficient curve, this selection of models is interesting to keep in an ensemble. 

![image](https://user-images.githubusercontent.com/1019791/73679335-7ccea400-46ba-11ea-8024-6ec812a900b1.png)

The idea is to have the same functionality of a `GridSearch` *but* we allow the search to figure out the pareto efficient frontier to select a subset of all the trained models. Here's an API:

```
ParetoGridSearch(
  estimatore=pipeline, 
  cv=10, 
  scores={'precision': precision, 'recall': recall, 'accuracy': accuracy}
  selectors=['precision', 'recall'] # track accuracy but don't subset based on it,
  reweight="average" # tactic for averaging the selected models at inference 
)
```
## Thoughts Behind It 

This does not solve all the possible ways that you might want to select from a set of models (maybe you want to select on the difference between `train` and `test` performance) but it does seem to be rather general. I am interested in the effect that this might have on mitigating drift. By having an average here you might just prevent over-fitting on your offline data, possibly making yourself a bit more robust against offline data.
